src/xattrs.c

```diff
@@ -37,6 +37,14 @@ enum xattrs_action_type_t_ {
 
 typedef enum xattrs_action_type_t_ xattrs_action_type_t;
 
+/* If SWUPD_WITH_XATTRS is not defined, create functions to override
+ * glibc's interface to the system calls
+ */
+#ifndef SWUPD_WITH_XATTRS
+#define lgetxattr(p,n,b,l) (-1)
+#define llistxattr(p,b,l) (0)
```

Comment:
While I think ENOTSUP makes sense in the error case, I'm more in favor of returning 0 (no xattrs to report) when it's not defined, as we don't care at that point and want to continue with the update.